ui/src/app/customizations/tech-extensions/import/tech-extension-import.component.ts

```diff
@@ -1,15 +1,111 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, ElementRef, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import {
+  FileUploader,
+  FileItem,
+  FileLikeObject,
+  FilterFunction,
+  FileUploaderOptions,
+  ParsedResponseHeaders
+} from 'ng2-file-upload-base/src';
+import { NotificationType } from 'patternfly-ng';
+import { Extension } from '../../../model';
 import { ExtensionStore } from '../../../store/extension/extension.store';
+import { NotificationService } from 'app/common/ui-patternfly/notification-service';
+
+interface FileError {
+  level: string;
+  message: string;
+}
 
 @Component({
   selector: 'syndesis-tech-extentions-import',
   templateUrl: 'tech-extension-import.component.html'
 })
+export class TechExtensionImportComponent {
+
+  uploader: FileUploader;
+  response: Extension = undefined;
+  error: FileError = undefined;
```

Comment:
Why are these fields initialized as `undefined`? I've seen this pattern repeated all over the app and I still cannot wrap my head around on why we are doing this.

Reply:
Habit, will take that out :-)